source/val/validate_type.cpp

```diff
@@ -165,6 +165,11 @@ spv_result_t ValidateTypeStruct(ValidationState_t& _, const Instruction* inst) {
       return _.diag(SPV_ERROR_INVALID_ID, inst)
              << "Structures cannot contain a void type.";
     }
+    if (member_type->opcode() == SpvOpTypeBool) {
```

Comment:
This is not the right place for this check. It is valid for a OpTypeBool to be a member of OpTypeStruct, but if it is then there are restrictions on which storage classes it may be used with.

Reply:
Updated the code to check "OpTypeBool can be used only with non-externally visible shader Storage Classes: Workgroup, CrossWorkgroup, Private, and Function.".